clab/clab.go

```diff
@@ -88,6 +88,12 @@ func WithRuntime(name string, d bool, dur time.Duration, gracefulShutdown bool)
 	}
 }
 
+func WithKeepMgmtNet() ClabOption {
+	return func(c *CLab) {
+		c.Runtime.WithKeepMgmtNet()
```

Comment:
do we need it to be a function? maybe just set a field on RuntimeConfig struct to `true`?

Reply:
@karimra this seemed to me to be the most in line approach... what do you think?

Reply:
Both approaches are fine, I added the `WithConfig` methods to avoid having a method per option.
At the end, it's about whether you want to allow changing this value after it's set or not.
At this moment, when clab is used only by the cli, both approaches are the same.

Reply:
To clarify, I am talking about the extra runtime interface method called `c.Runtime.WithKeepMgmtNet()`
to me it looks like doing this provides less boilerplate:

```go
func WithKeepMgmtNet() ClabOption {
	return func(c *CLab) {
             c.Runtime.keepMgmtNet = true
        }
}
```
and we keep interface smaller at the same time


Another approach is to use the existing `WithConfig` interface method and set the runtimeconfig's `keepMgmtNet` field there

Reply:
You can't do it that way, `Runtime` is an interface, it doesn't have fields, only methods.

What I'm talking about is adding  ` KeepMgmtNet bool` to:
```golang
type RuntimeConfig struct {
	Timeout          time.Duration
	GracefulShutdown bool
	Debug            bool
}
```

Reply:
@karimra which would result in adding that parameter to withConfig, which would require to specify this parameter even in deploy or inspect scenarios something that I found disturbing ...

Reply:
No need to specify it, the default value is enough in those cases.
anyways, this is just a matter of style/preference.